tests/unit-tests/shell/test_basic_idle_handler.cpp

```diff
@@ -0,0 +1,204 @@
+    MOCK_METHOD1(set_power_mode, void(MirPowerMode));
+};
+
+struct BasicIdleHandler: Test
+{
+    NiceMock<mtd::MockIdleHub> idle_hub;
+    mtd::StubInputScene input_scene;
+    mtd::StubBufferAllocator allocator;
+    StrictMock<MockDisplayConfigurationController> display;
+};
+}
+
+TEST_F(BasicIdleHandler, does_not_register_observers_by_default)
+{
+    EXPECT_CALL(idle_hub, register_interest(_, _))
+        .Times(0);
+    msh::BasicIdleHandler handler{
+        mt::fake_shared(idle_hub),
+        mt::fake_shared(input_scene),
+        mt::fake_shared(allocator),
+        mt::fake_shared(display)};
+}
+
+TEST_F(BasicIdleHandler, turns_display_off_when_idle)
+{
+    std::weak_ptr<ms::IdleStateObserver> observer;
+    msh::BasicIdleHandler handler{
+        mt::fake_shared(idle_hub),
+        mt::fake_shared(input_scene),
+        mt::fake_shared(allocator),
+        mt::fake_shared(display)};
+    EXPECT_CALL(idle_hub, register_interest(_, _))
+        .Times(AnyNumber());
+    EXPECT_CALL(idle_hub, register_interest(_, Eq(30s)))
+        .WillOnce(SaveArg<0>(&observer));
+    handler.set_display_off_timeout(30s);
+    auto const locked = observer.lock();
+    ASSERT_THAT(locked, Ne(nullptr));
+    EXPECT_CALL(display, set_power_mode(mir_power_mode_off));
+    locked->idle();
+    EXPECT_CALL(display, set_power_mode(mir_power_mode_on)).Times(AnyNumber());
+}
+
+TEST_F(BasicIdleHandler, turns_display_back_on_when_active)
+{
+    std::weak_ptr<ms::IdleStateObserver> observer;
+    msh::BasicIdleHandler handler{
+        mt::fake_shared(idle_hub),
+        mt::fake_shared(input_scene),
+        mt::fake_shared(allocator),
+        mt::fake_shared(display)};
+    EXPECT_CALL(idle_hub, register_interest(_, _))
+        .Times(AnyNumber());
+    EXPECT_CALL(idle_hub, register_interest(_, Eq(30s)))
+        .WillOnce(SaveArg<0>(&observer));
+    handler.set_display_off_timeout(30s);
+    auto const locked = observer.lock();
+    ASSERT_THAT(locked, Ne(nullptr));
+    EXPECT_CALL(display, set_power_mode(mir_power_mode_off));
+    locked->idle();
+    EXPECT_CALL(display, set_power_mode(mir_power_mode_on));
+    locked->active();
+    EXPECT_CALL(display, set_power_mode(mir_power_mode_on)).Times(AnyNumber());
+}
+
+TEST_F(BasicIdleHandler, off_timeout_is_disabled_on_destruction)
+{
+    std::weak_ptr<ms::IdleStateObserver> observer;
+    msh::BasicIdleHandler handler{
+        mt::fake_shared(idle_hub),
+        mt::fake_shared(input_scene),
+        mt::fake_shared(allocator),
+        mt::fake_shared(display)};
+    EXPECT_CALL(idle_hub, register_interest(_, _))
+        .Times(AnyNumber());
+    EXPECT_CALL(idle_hub, register_interest(_, Eq(30s)))
+        .WillOnce(SaveArg<0>(&observer));
+    handler.set_display_off_timeout(30s);
+    auto const locked = observer.lock();
+    ASSERT_THAT(locked, Ne(nullptr));
+    EXPECT_CALL(idle_hub, unregister_interest(_))
+        .Times(AnyNumber());
```

Comment:
It takes a careful reading to see that this is not part of the setup, but filtering "uninteresting" events. Maybe this would be clearer:
```suggestion
    EXPECT_CALL(idle_hub, unregister_interest(Not(Ref(*locked))))
        .Times(AnyNumber());
```